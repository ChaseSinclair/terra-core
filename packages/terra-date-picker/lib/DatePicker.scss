@import '~terra-mixins';
@import './variables';
@import './mixins';
@import '../node_modules/react-datepicker/dist/react-datepicker.min';

.terra-DatePicker {
  display: inline-block;
  position: relative;
}

.terra-DatePicker-input {
  border: 1px solid #aeaeae;
  border-radius: 4px;
  box-shadow: inset 0 2px 2px #e9e9e9;
  font-size: 13px;
  line-height: 16px;
  margin-top: 0;
  padding: 6px 10px 5px;

  &:focus {
    border-color: $terra-blue-60;
    box-shadow: inset 0 2px 2px #e9e9e9, 0 0 10px 0 rgba(73, 107, 125, 0.3);
    outline: none;
  }
}

/* ignore the selector-class-pattern rule for selector classes in third party components */
/* stylelint-disable selector-class-pattern */
.react-datepicker__day.react-datepicker__day--highlighted {
  // background-color: $terra-secondary-color-50;
  border-radius: 0.3rem;
  color: $terra-date-picker-day-color-highlighted; 
}

.react-datepicker__day--highlighted:hover {
  // background-color: $terra-secondary-color-60; 
}

.react-datepicker__day.react-datepicker__day--keyboard-selected,
.react-datepicker__day.react-datepicker__day--selected,
.react-datepicker__day.react-datepicker__day--in-range,
.react-datepicker__day.react-datepicker__day--in-selecting-range {
  color: $terra-date-picker-day-color-selected;
}

.react-datepicker__day {
  color: $terra-date-picker-day-color;
}

.react-datepicker__day--outside-month {
  color: $terra-date-picker-day-color-outside-month;
}

.react-datepicker__day--disabled {
  color: $terra-date-picker-day-color-disabled;
}
/* stylelint-enable */


