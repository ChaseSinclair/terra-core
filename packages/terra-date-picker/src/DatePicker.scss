@import '~terra-mixins';
:global { @import '~react-datepicker/dist/react-datepicker.min'; }

.datePicker {
  display: inline-block;
  min-width: 14.643em;
}

.input {
  @include terra-border-radius-start(0.286em);
  @include terra-margin-end(-0.071em);
  border: 0.071em solid #a7aaab;
  box-shadow: inset 0 0.143em 0.143em #d3d4d5;
  font-size: 1em;
  line-height: 1.143;
  margin: 0.357em 0;
  padding: 0.429em 0.714em;
  vertical-align: middle;

  &:focus {
    border-color: #26a2e5;
    outline: none;
  }
}

.button {
  @include terra-border-radius-start(0);
  @include terra-margin-start(-0.071em);
}
