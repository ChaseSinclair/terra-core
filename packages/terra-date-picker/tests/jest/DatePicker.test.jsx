import React from 'react';
import moment from 'moment';
import DatePicker from '../../src/DatePicker';
import DateRange from '../../src/DateRange';

const isWeekday = (date) => {
  const day = date.day();
  return day !== 0 && day !== 6;
};

it('should render a default date input and date picker', () => {
  const datePicker = <DatePicker />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date range', () => {
  const dateRange = <DateRange />;
  const wrapper = shallow(dateRange);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date picker with disabled dates', () => {
  const datePicker = <DatePicker excludeDates={[moment().subtract(1, 'days'), moment().add(1, 'days')]} />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date picker with filtered dates', () => {
  const datePicker = <DatePicker filterDate={isWeekday} />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date picker with highlighted dates', () => {
  const datePicker = <DatePicker highlightDates={[moment().subtract(3, 'days'), moment().add(3, 'days')]} />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date picker with included dates', () => {
  const datePicker = <DatePicker includeDates={[moment(), moment().subtract(1, 'days'), moment().add(1, 'days')]} />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date picker with min and max dates', () => {
  const datePicker = <DatePicker minDate={moment()} maxDate={moment().add(6, 'days')} />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date picker with two months', () => {
  const datePicker = <DatePicker monthsShown={2} />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date picker that opens to a specific date', () => {
  const datePicker = <DatePicker openToDate={moment('1999-09-09')} />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date picker with the month and year dropdowns hidden', () => {
  const datePicker = <DatePicker showMonthDropdown={false} showYearDropdown={false} />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date picker with the Today button', () => {
  const datePicker = <DatePicker todayButton="Today123" />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date picker with the Today button hidden', () => {
  const datePicker = <DatePicker hideTodayButton />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});

it('should render a date picker in portal', () => {
  const datePicker = <DatePicker portal />;
  const wrapper = shallow(datePicker);
  expect(wrapper).toMatchSnapshot();
});
