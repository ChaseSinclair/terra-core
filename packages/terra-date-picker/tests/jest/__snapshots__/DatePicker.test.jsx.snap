exports[`test should render a date picker in portal 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar=" "
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    fixedHeight={true}
    locale="en-US"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    portal={true}
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={true}
    showYearDropdown={true}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    todayButton="Today"
    utcOffset={-360}
    withPortal={false} />
</div>
`;

exports[`test should render a date picker that opens to a specific date 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar=" "
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    fixedHeight={true}
    locale="en-US"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    openToDate={"1999-09-09T05:00:00.000Z"}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={true}
    showYearDropdown={true}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    todayButton="Today"
    utcOffset={-360}
    withPortal={false} />
</div>
`;

exports[`test should render a date picker with disabled dates 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar=" "
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    excludeDates={
      Array [
        "2017-03-09T21:29:19.380Z",
        "2017-03-11T21:29:19.381Z",
      ]
    }
    fixedHeight={true}
    locale="en-US"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={true}
    showYearDropdown={true}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    todayButton="Today"
    utcOffset={-360}
    withPortal={false} />
</div>
`;

exports[`test should render a date picker with filtered dates 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar=" "
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    filterDate={[Function]}
    fixedHeight={true}
    locale="en-US"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={true}
    showYearDropdown={true}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    todayButton="Today"
    utcOffset={-360}
    withPortal={false} />
</div>
`;

exports[`test should render a date picker with highlighted dates 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar=" "
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    fixedHeight={true}
    highlightDates={
      Array [
        "2017-03-07T21:29:19.386Z",
        "2017-03-13T20:29:19.386Z",
      ]
    }
    locale="en-US"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={true}
    showYearDropdown={true}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    todayButton="Today"
    utcOffset={-360}
    withPortal={false} />
</div>
`;

exports[`test should render a date picker with included dates 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar=" "
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    fixedHeight={true}
    includeDates={
      Array [
        "2017-03-10T21:29:19.389Z",
        "2017-03-09T21:29:19.389Z",
        "2017-03-11T21:29:19.389Z",
      ]
    }
    locale="en-US"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={true}
    showYearDropdown={true}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    todayButton="Today"
    utcOffset={-360}
    withPortal={false} />
</div>
`;

exports[`test should render a date picker with min and max dates 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar=" "
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    fixedHeight={true}
    locale="en-US"
    maxDate={"2017-03-16T20:29:19.392Z"}
    minDate={"2017-03-10T21:29:19.392Z"}
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={true}
    showYearDropdown={true}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    todayButton="Today"
    utcOffset={-360}
    withPortal={false} />
</div>
`;

exports[`test should render a date picker with the Today button 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar=" "
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    fixedHeight={true}
    locale="en-US"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={true}
    showYearDropdown={true}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    todayButton="Today123"
    utcOffset={-360}
    withPortal={false} />
</div>
`;

exports[`test should render a date picker with the Today button hidden 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar=" "
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    fixedHeight={true}
    locale="en-US"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={true}
    showYearDropdown={true}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    utcOffset={-360}
    withPortal={false} />
</div>
`;

exports[`test should render a date picker with the month and year dropdowns hidden 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar="MMMM YYYY"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    fixedHeight={true}
    locale="en-US"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={false}
    showYearDropdown={false}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    todayButton="Today"
    utcOffset={-360}
    withPortal={false} />
</div>
`;

exports[`test should render a date picker with two months 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar=" "
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    fixedHeight={true}
    locale="en-US"
    monthsShown={2}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={true}
    showYearDropdown={true}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    todayButton="Today"
    utcOffset={-360}
    withPortal={false} />
</div>
`;

exports[`test should render a date range 1`] = `
<div
  className="terra-DatePicker-range">
  <DatePicker
    hideTodayButton={false}
    isEndDateRange={false}
    isStartDateRange={true}
    monthsShown={1}
    onChange={[Function]}
    withPortal={false} />
  <DatePicker
    hideTodayButton={false}
    isEndDateRange={true}
    isStartDateRange={false}
    monthsShown={1}
    onChange={[Function]}
    withPortal={false} />
</div>
`;

exports[`test should render a default date input and date picker 1`] = `
<div
  className="terra-DatePicker">
  <DatePicker
    className="terra-DatePicker-input"
    dateFormat="MM/DD/YYYY"
    dateFormatCalendar=" "
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    fixedHeight={true}
    locale="en-US"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onMonthChange={[Function]}
    placeholderText="MM/DD/YYYY"
    popoverAttachment="top left"
    popoverTargetAttachment="bottom left"
    popoverTargetOffset="10px 0"
    selectsEnd={false}
    selectsStart={false}
    showMonthDropdown={true}
    showYearDropdown={true}
    tetherConstraints={
      Array [
        Object {
          "attachment": "together",
          "to": "window",
        },
      ]
    }
    todayButton="Today"
    utcOffset={-360}
    withPortal={false} />
</div>
`;
